elasticsearch/src/main/java/org/hibernate/search/elasticsearch/client/impl/DefaultElasticsearchClientFactory.java

```diff
@@ -81,16 +84,26 @@ private Sniffer createSniffer(String scopeName, RestClient client, Properties pr
 				ElasticsearchEnvironment.Defaults.DISCOVERY_ENABLED
 		);
 		if ( discoveryEnabled ) {
-			return Sniffer.builder( client )
+			SnifferBuilder builder = Sniffer.builder( client )
 					.setSniffIntervalMillis(
 							ConfigurationParseHelper.getIntValue(
 									properties,
 									propertyPrefix + ElasticsearchEnvironment.DISCOVERY_REFRESH_INTERVAL,
 									ElasticsearchEnvironment.Defaults.DISCOVERY_REFRESH_INTERVAL
 							)
 							* 1_000 // The configured value is in seconds
-					)
-					.build();
+					);
+			String scheme = ConfigurationParseHelper.getString(properties, propertyPrefix + ElasticsearchEnvironment.DISCOVERY_SCHEME, "http");
+
+			// https discovery support
+			if ( scheme.equals(ElasticsearchHostsSniffer.Scheme.HTTPS.toString()) ) {
+				HostsSniffer hostsSniffer = new ElasticsearchHostsSniffer(
+						client,
+						ElasticsearchEnvironment.Defaults.DISCOVERY_REFRESH_INTERVAL * 1_000,
```

Comment:
as you prefer. Typically we just force-push on the same branch, that will replace the current code in the same PR.